lib/install/management/dispatcher.go

```diff
@@ -245,3 +254,47 @@ func (d *Dispatcher) CollectDiagnosticLogs() {
 		}
 	}
 }
+
+func (d *Dispatcher) DeleteVCH(conf *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infof("Removing VMs")
+
+	if err := d.DeleteVCHInstances(conf); err != nil {
+		return err
+	}
+	if err := d.destroyResourcePoolIfEmpty(conf); err != nil {
+		return err
+	}
+	return d.removeNetwork(conf)
+}
+
+func (d *Dispatcher) DeleteVCHInstances(conf *metadata.VirtualContainerHostConfigSpec) error {
+	var errs []string
+
+	var err error
+	var vmm *vm.VirtualMachine
+	var children []*vm.VirtualMachine
+
+	if vmm, err = d.findAppliance(conf); err != nil {
+		return err
+	}
+	if vmm == nil {
```

Comment:
Apologies - you're entirely correct. We would need a --really-force flag for that :)

We should ensure that the appliance is the last VM deleted in that case - that would mean that if delete fails for part way through deleting children and some are left behind we should still be able to run delete again against that VCH.